src/core/reference/include/openvino/reference/identity.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstring>
+
+#include "openvino/core/shape.hpp"
+
+namespace ov {
+namespace reference {
+namespace identity {
+
+/**
+ * @brief Identity operation computes the identity of the input tensor.
+ *
+ * @param input Input matrix (matrices) pointer.
+ * @param output Output matrix (matrices) pointer.
+ * @param copy Boolean that determines whether to return the input as output or
+ * copy the input to a new memory address.
+ **/
+template <typename T>
+void identity(const T** input, T** output, const Shape& shape, const bool copy) {
+    const auto total_elements = shape_size<Shape>(shape);
+
+    if (!copy) {
+        *output = *input;
+    } else {
+        std::memcpy(*output, *input, total_elements * sizeof(T));
+    }
```

Comment:
The motivation to optimize copying by reusing the pointer is understandable, but it should be implemented carefully as the input ptr is const and the output is not. 
Also to copy the data, the size of the type can be used directly instead of templating the whole function, for example:
https://github.com/openvinotoolkit/openvino/blob/cda1cbd54c4e428f06c99eac1538a048f1665392/src/core/reference/include/openvino/reference/reshape.hpp#L22-L24 Additionally for the reference to consider, the Identity op with evaluate is used as an example of op extension in docs:
https://github.com/openvinotoolkit/openvino/blob/cda1cbd54c4e428f06c99eac1538a048f1665392/src/core/template_extension/identity.cpp#L36-L45

Reply:
Idea applied!

Reply:
I can see in the current version of the spec that the output is always a "copy". Is that guarantee needed?
Because in some cases like:
https://github.com/openvinotoolkit/openvino/blob/cda1cbd54c4e428f06c99eac1538a048f1665392/src/core/template_extension/identity.cpp#L40-L41
Why not apply this simple optimization?

Reply:
Added to reference!